discover/gpu.go

```diff
@@ -517,11 +516,15 @@ func FindGPULibs(baseLibName string, defaultPatterns []string) []string {
 	slog.Debug("Searching for GPU library", "name", baseLibName)
 
 	// Start with our bundled libraries
-	patterns := []string{}
-	for _, d := range LibraryDirs() {
-		patterns = append(patterns, filepath.Join(d, baseLibName))
+	exe, err := os.Executable()
+	if err != nil {
+		return []string{}
 	}
 
+	libDir := filepath.Join(filepath.Dir(exe), envconfig.LibRelativeToExe(), "lib", "ollama")
```

Comment:
maybe out of scope but this isn't actually configured by the environment. consider merging it into this package

Reply:
Was thinking the same thing. It's used in a few places so I'll revisit in a follow up since I think most of them can be eliminated